src/components/ClockFace.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import classnames from 'classnames';
+import SettingsModal from './SettingsModal';
+
+const ClockFace = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isActive, setIsActive] = useState(false);
+  const [currentSessionIndex, setCurrentSessionIndex] = useState(0);
+  // in seconds
+  const [sessions, setSessions] = useState([60, 5, 10, 10]);
+  const [currentSeconds, setCurrentSeconds] = useState(sessions[currentSessionIndex]);
+
+  useEffect(() => {
+    let timeInterval;
+    if (isActive) {
+      timeInterval = setInterval(() => {
+        if (currentSeconds > 0) {
+          setCurrentSeconds(currentSeconds - 1);
+        } else {
+          setIsActive(false);
+          clearInterval(timeInterval);
+        }
+      }, 1000)
+    }
+    else if (!isActive) {
+      clearInterval(timeInterval)
+    }
+    return () => clearInterval(timeInterval)
+  }, [isActive, currentSeconds]);
+
+  useEffect(() => {
+    if (isActive && currentSeconds === 0) {
+      setCurrentSessionIndex(currentSessionIndex + 1);
+    }
+    else if (currentSessionIndex === sessions.length) {
+      setCurrentSessionIndex(0);
+      setIsActive(false)
+    }
+  }, [currentSeconds, isActive])
+
+  useEffect(() => {
+    setCurrentSeconds(sessions[currentSessionIndex])
+  }, [currentSessionIndex])
+
+  useEffect(() => {
+    if (isModalOpen === true) {
+      setIsActive(false);
+    }
+  }, [isModalOpen])
+
+  return (
+    <div className='ClockFace'>
+      <div className='ClockFace__info'>
+        <h3>I am a Pomodoro Timer</h3>
+        <h4>Current Session: {currentSessionIndex + 1}</h4>
+        <h4
+          className={classnames('timer', { isActive })}
+        >
+          00 :
+          {Math.floor(currentSeconds / 60) < 10 ? `0${Math.floor(currentSeconds / 60)}` : Math.floor(currentSeconds / 60)} :
```

Comment:
Move this logic into a variable instead. You should have as few logic in jsx as possible. There's string formatting library you can use.

Reply:
Do you mean [react-string-format ](https://www.npmjs.com/package/react-string-format)...? But I dont know how it can help with making this easier.
This makes no difference:
```
  const formattedMinutes = Math.floor(currentSeconds / 60) < 10 ? `0${Math.floor(currentSeconds / 60)}` : Math.floor(currentSeconds / 60);
  const formattedSeconds = currentSeconds % 60 < 10 ? `0${currentSeconds % 60}` : currentSeconds % 60
```
So instead I can do: 
```
  const formattedMinutes = Math.floor(currentSeconds / 60);
  const formattedSeconds = currentSeconds % 60;
...
         <h4>
          00 :
          {formattedMinutes < 10 ? `0${formattedMinutes}` : formattedMinutes} :
          {formattedSeconds < 10 ? `0${formattedSeconds}` : formattedSeconds}
        </h4>
```

Reply:
> This makes no difference:
> 
> ```
>   const formattedMinutes = Math.floor(currentSeconds / 60) < 10 ? `0${Math.floor(currentSeconds / 60)}` : Math.floor(currentSeconds / 60);
>   const formattedSeconds = currentSeconds % 60 < 10 ? `0${currentSeconds % 60}` : currentSeconds % 60
> ```

This is actually way better to read in the JSX IMO. JSX should have as few custom logic as possible. If you extract these more complex operation out and make it become a variable as a user I don't need to scratch my head trying to figure out what these two lines are trying to achieve. 
```js
00 : {formattedMinutes} : {formattedSeconds}
```